src/table/index.js

```diff
@@ -158,29 +133,32 @@ class TableField extends Component {
     this.props.onChange(removePosition(filteredRows));
   }
   handleRowSelect(row, isSelected, e) {
+    const { onChange } = this.props;
+
     const {
-      data,
+      data = [],
       selectRow: { onSelectRow: { fieldToUpdate = "picked" } }
     } = this.tableConf;
-    let filteredRows = (data || []).map(item => {
-      if (!isSelected && item[fieldToUpdate] !== undefined) {
-        if (isEquivalentObject(item, row)) {
-          delete item[fieldToUpdate];
-        }
-      } else if (isEquivalentObject(item, row)) {
-        item[fieldToUpdate] = isSelected;
-      }
-      return item;
-    });
-    this.props.onChange(filteredRows);
+
+    row.isSelected = isSelected;
+
+    if (!isSelected) {
+      delete row[fieldToUpdate];
+    } else {
+      row[fieldToUpdate] = isSelected;
+    }
+
+    onChange(data);
   }
+
   handleAllRowSelect(isSelected, rows, e) {
     const {
-      // data,
+      data,
       selectRow: { onSelectAllRow: { fieldToUpdate = "picked" } }
     } = this.tableConf;
 
-    let filteredRows = (rows || []).map(item => {
+    let filteredRows = data.map(item => {
+      item["isSelected"] = isSelected;
```

Comment:
i think we have used isSelected index in the plugin sync.(patient dashboard sync).
 please make sure its not affect the existing functionality.

Reply:
Switch property name